### PR TITLE
RecordValue.ToExpression to escape identifiers

### PR DESCRIPTION
**Pull request by anderson-joyle**
Issue https://github.com/microsoft/Power-Fx/issues/1409.

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Lexer/TexlLexer.cs [hunk 1]
```diff
@@ -246,6 +246,11 @@ public static TexlLexer GetLocalizedInstance(CultureInfo culture)
         public static IReadOnlyList<string> GetKeywords()
         {
             return _keywords.Keys.ToList();
+        }
+
+        public static IReadOnlyList<string> GetReservedKeywords()
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Lexer/TexlLexer.cs:251]: Ienumerable? then we can avoid ToList call.

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/SerializationTests.cs [hunk 2]
```diff
@@ -97,7 +100,44 @@ public void OptionSetDefaultExpressionValueErrorTests()
...
+        }
+
+        [Fact]
+        public void RecordValueKeyworkSerializationTest()
+        {
+            var engine = new RecalcEngine(new PowerFxConfig());
+            var symbol = new SymbolTable();
+
+            // Combining both keywords and reserved keywords
+            var keywords = TexlLexer.GetKeywords().Concat(TexlLexer.GetReservedKeywords()).ToList();
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Interpreter.Tests/SerializationTests.cs:112]: this list should be fixed. can we inline it?

File: src/libraries/Microsoft.PowerFx.Core/Lexer/TexlLexer.cs (hunk 1 above)
Comment (anderson-joyle) [reply src/libraries/Microsoft.PowerFx.Core/Lexer/TexlLexer.cs]: This has been removed since we adopted the inline approach in the new test cases.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by MikeStall — APPROVED** [review]

**Pull request merged**